### PR TITLE
Create log configuration functions

### PR DESCRIPTION
**Pull request by CaioCordeiro**
Changes:

- Create src/log.c and src/log.h and Add them to src/Makefile.am;
- Create ell lib configuration functions;
- Include log.h in knot_cloud.c;
- Setup log configurations inside knot_cloud_start function;
- Create knot_cloud_set_log_priority, this function lets the user
choose the SDK log priority level;
- Add function to knot_cloud.h

### CODE REVIEW

File: src/log.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
```
Comment (luigiluz) [inline src/log.c]:
```suggestion
 * Copyright (c) 2020, CESAR. All rights reserved.
```

File: src/log.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+{
+	l_log_set_handler(log_stderr_handler);
+}
+
+void log_set_priority(int priority)
+{
+    log_priority = priority;    
+    if (log_priority == L_LOG_DEBUG)
+		l_debug_enable("*");
+}
```
Comment (luigiluz) [inline src/log.c]: Please add a blank line at the end of the file

File: src/log.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
```
Comment (luigiluz) [inline src/log.c:21]:
```suggestion

/**
 *  Log source file
 */
 
```

File: src/log.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2019, CESAR. All rights reserved.
```
Comment (luigiluz) [inline src/log.h]:
```suggestion
 * Copyright (c) 2020, CESAR. All rights reserved.
```

File: src/log.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
```
Comment (luigiluz) [inline src/log.h:21]:
```suggestion

/**
 *  Log header file
 */
 
```

File: src/log.h [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+void log_set_priority(int priority);
+void log_ell_enable();
```
Comment (luigiluz) [inline src/log.h]: Please add a blank line at the end of the file

**Review by luigiluz — CHANGES_REQUESTED** [review]

File: src/knot_cloud.c [hunk 7]
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
 	return mq_start(url, connected_cb, disconnected_cb, user_data);
 }
 
+void knot_cloud_set_log_priority(char * priority)
+{
+    if (!strcmp(priority,"error"))
```
Comment (alla3) [inline src/knot_cloud.c]: This indentation is using spaces instead of tab. Please, change it.

File: src/knot_cloud.c [hunk 8]
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
...
+		log_set_priority(L_LOG_ERR);
+	else if (!strcmp(priority,"warn"))
+		log_set_priority(L_LOG_WARNING);
+	else if (!strcmp(priority,"info"))
+		log_set_priority(L_LOG_INFO);
+	else if (!strcmp(priority,"debug"))
+		log_set_priority(L_LOG_DEBUG);
+}
```
Comment (alla3) [inline src/knot_cloud.c]: This function is not following the same order of the header.

File: src/log.c [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+{
+	l_log_set_handler(log_stderr_handler);
+}
+
+void log_set_priority(int priority)
+{
+    log_priority = priority;    
+    if (log_priority == L_LOG_DEBUG)
+		l_debug_enable("*");
+}
```
Comment (alla3) [inline src/log.c]: I would add this function on commit 9944be4700a2e2088b1afc89e85da0204767ffd1, since it is not been used on this commit.

**Review by alla3 — CHANGES_REQUESTED** [review]

File: src/knot_cloud.c [hunk 10]
```diff
@@ -723,6 +723,18 @@ int knot_cloud_start(char *url, char *user_token,
 	return mq_start(url, connected_cb, disconnected_cb, user_data);
 }
 
+void knot_cloud_set_log_priority(char * priority)
```
Comment (alla3) [inline src/knot_cloud.c]:
```suggestion
void knot_cloud_set_log_priority(char *priority)
```

File: src/knot_cloud.h [hunk 11]
```diff
@@ -52,12 +52,14 @@ typedef bool (*knot_cloud_cb_t) (const struct knot_cloud_msg *msg,
...
 typedef void (*knot_cloud_disconnected_cb_t) (void *user_data);
 
+
 int knot_cloud_read_start(const char *id, knot_cloud_cb_t read_handler_cb,
 			  void *user_data);
 int knot_cloud_start(char *url, char *user_token,
 		     knot_cloud_connected_cb_t connected_cb,
 		     knot_cloud_disconnected_cb_t disconnected_cb,
 		     void *user_data);
+void knot_cloud_set_log_priority(char * priority);
```
Comment (alla3) [inline src/knot_cloud.h]:
```suggestion
void knot_cloud_set_log_priority(char *priority);
```

File: src/log.c [hunk 12]
```diff
@@ -63,3 +63,9 @@ void log_ell_enable()
 {
 	l_log_set_handler(log_stderr_handler);
 }
+void log_set_priority(int priority)
```
Comment (alla3) [inline src/log.c:62]: Please, add a blank line between functions.

Comment (vitbaq) [inline src/log.c:62]: Put a space before this line

**Review by vitbaq — CHANGES_REQUESTED** [review]

**Review by alla3 — CHANGES_REQUESTED** [review]

File: src/log.c [hunk 13]
```diff
@@ -63,3 +63,10 @@ void log_ell_enable()
 {
 	l_log_set_handler(log_stderr_handler);
 }
+
+void log_set_priority(int priority)
```
Comment (vitbaq) [inline src/log.c:62]: Change the order of the functions. Put this function after the log_ell_enable

File: src/log.c [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+		fprintf(stderr, "DEBUG: ");
+		break;
+	default:
+		return;
+	}
+
+	vfprintf(stderr, format, ap);
+}
+
+void log_ell_enable()
```
Comment (vitbaq) [inline src/log.c]: Change to `void log_ell_enable(void)`

File: src/log.h [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+/**
+ * Log header file
+ */
+
+void log_ell_enable();
```
Comment (vitbaq) [inline src/log.h]: Change to `void log_ell_enable(void)`

**Review by vitbaq — CHANGES_REQUESTED** [review]
Almost there, the last things to change.

File: src/log.c (hunk 14 above)
Comment (vitbaq) [reply src/log.c]: Reason for this: [link](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/287002)

**Review by vitbaq — APPROVED** [review]

**Review by alla3 — APPROVED** [review]

**Pull request merged**